source/crockford.ts

```diff
@@ -0,0 +1,58 @@
+// Code from https://github.com/devbanana/crockford-base32/blob/develop/src/index.ts
+import { B32_CHARACTERS } from "./constants.js";
+
+export function crockfordEncode(input: Buffer): string {
+    const output: number[] = [];
+    let bitsRead = 0;
+    let buffer = 0;
+
+    const reversedInput = Buffer.from(input).reverse();
+
+    for (const byte of reversedInput) {
+        buffer |= byte << bitsRead;
+        bitsRead += 8;
+
+        while (bitsRead >= 5) {
+            output.unshift(buffer & 0x1f);
+            buffer >>>= 5;
+            bitsRead -= 5;
+        }
+    }
+
+    if (bitsRead > 0) {
+        output.unshift(buffer & 0x1f);
+    }
+
+    return output.map(byte => B32_CHARACTERS.charAt(byte)).join("");
+}
+
+export function crockfordDecode(input: string): Buffer {
+    let sanitizedInput = input.toUpperCase().split("").reverse().join("");
+
+    const output: number[] = [];
+    let bitsRead = 0;
+    let buffer = 0;
+
+    for (const character of sanitizedInput) {
+        const byte = B32_CHARACTERS.indexOf(character);
+
+        if (byte === -1) {
+            throw new Error(`Invalid base 32 character found in string: ${character}`);
+        }
+
+        buffer |= byte << bitsRead;
+        bitsRead += 5;
+
+        while (bitsRead >= 8) {
+            output.unshift(buffer & 0xff);
+            buffer >>>= 8;
+            bitsRead -= 8;
+        }
+    }
+
+    if (bitsRead >= 5 || buffer > 0) {
+        output.unshift(buffer & 0xff);
+    }
+
+    return Buffer.from(output);
```

Comment:
Buffer isn't available in the browser.. and I'd prefer to avoid using it if at all possible. ArrayBuffer and Uint* style arrays are fine however.

Reply:
Sure, I replaced it with `Uint8Array`.